tests/unit/test_state.py

```diff
@@ -118,6 +118,16 @@ def test_change_state_persists(state_instance):
 
         assert d.get('created')
 
+def test_change_state_deps_installed(state_instance):
+    assert not state_instance.created
+
```

Comment:
I think we are okay with just the above.  We don't need the rest of the test.  We already have tests that we can set/get through `change_state`.